app/src/androidTest/java/com/github/se/cyrcle/MainActivityTest.kt

```diff
@@ -0,0 +1,163 @@
+import androidx.compose.ui.test.junit4.ComposeTestRule
+import androidx.compose.ui.test.junit4.createAndroidComposeRule
+import androidx.compose.ui.test.onChildAt
+import androidx.compose.ui.test.onNodeWithTag
+import androidx.compose.ui.test.performClick
+import androidx.compose.ui.test.performScrollToIndex
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.github.se.cyrcle.ui.navigation.TopLevelDestinations
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class MainActivityTest {
+
+  @get:Rule val composeTestRule = createAndroidComposeRule<MainActivity>()
+
+  private lateinit var authRobot: AuthScreenRobot
+  private lateinit var mapRobot: MapScreenRobot
+  private lateinit var listRobot: ListScreenRobot
+  private lateinit var cardRobot: CardScreenRobot
+
+  @Before
+  fun setUp() {
+
+    // TODO inject a fake database for testing
+    // Currently, the test uses the real database
+
+    authRobot = AuthScreenRobot(composeTestRule)
+    mapRobot = MapScreenRobot(composeTestRule)
+    listRobot = ListScreenRobot(composeTestRule)
+    cardRobot = CardScreenRobot(composeTestRule)
+  }
+
+  @Test
+  fun reviewCardDisplaysWell() {
+    authRobot.assertAuthScreen()
+    authRobot.performSignIn()
+
+    mapRobot.assertMapScreen()
+    mapRobot.toList()
+
+    listRobot.assertListScreen()
+    listRobot.toCard(0)
+
+    cardRobot.assertCardScreen()
+  }
+
+  private class CardScreenRobot(val composeTestRule: ComposeTestRule) {
+    fun assertCardScreen() {
+      composeTestRule.onNodeWithTag("TopAppBar").assertIsDisplayed()
+      composeTestRule.onNodeWithTag("RowCapacityRack").assertIsDisplayed()
+      composeTestRule.onNodeWithTag("RowProtectionPrice").assertIsDisplayed()
+      composeTestRule.onNodeWithTag("RowSecurity").assertIsDisplayed()
+      composeTestRule.onNodeWithTag("ButtonsColumn").assertIsDisplayed()
+      composeTestRule.onNodeWithTag("ShowInMapButton").assertIsDisplayed().assertHasClickAction()
+      composeTestRule.onNodeWithTag("AddReviewButton").assertIsDisplayed().assertHasClickAction()
+      composeTestRule.onNodeWithTag("ReportButton").assertIsDisplayed().assertHasClickAction()
+    }
+
+    @OptIn(ExperimentalTestApi::class)
+    fun goBack() {
+      composeTestRule
+          .onNodeWithTag("GoBackButton")
+          .assertIsDisplayed()
+          .assertHasClickAction()
+          .performClick()
+      composeTestRule.waitUntilAtLeastOneExists(
+          hasTestTag("SpotListScreen").or(hasTestTag("MapScreen")))
+    }
+  }
+
+  private class ListScreenRobot(val composeTestRule: ComposeTestRule) {
+
+    @OptIn(ExperimentalTestApi::class)
+    fun toCard(index: Int) {
+      composeTestRule.onNodeWithTag("SpotListColumn").performScrollToIndex(index)
+      composeTestRule
+          .onNodeWithTag("SpotListColumn")
+          .onChildAt(index + 1)
+          .assertIsDisplayed()
```

Comment:
this assertion fails on my end for some reason. Let's double check with @andrepeiry to see if it's just me

Reply:
Could you maybe add a `Thread.sleep(10000)` and check on the emulator screen what the screen looks like at that point?